apps/src/componentLibrary/tooltip/WithTooltip.tsx

```diff
@@ -13,16 +30,180 @@ const WithTooltip: React.FunctionComponent<WithTooltipProps> = ({
   tooltipOverlayClassName,
   tooltipProps,
 }) => {
+  const [nodePosition, setNodePosition] = useState<HTMLElement | null>(null);
+  const [showTooltip, setShowTooltip] = useState<boolean>(false);
+  const [tooltipStyles, setTooltipStyles] = useState<React.CSSProperties>({});
+  const tooltipRef = useRef<HTMLDivElement | null>(null);
+
+  // Define the additional event handlers
+  const handleFocus = (event: React.FocusEvent<HTMLElement>) => {
+    setShowTooltip(true);
+    setNodePosition(event.target);
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLElement>) => {
+    setShowTooltip(false);
+    setNodePosition(null);
+  };
+
+  const handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
+    setShowTooltip(true);
+    setNodePosition(event.target as HTMLElement);
+  };
+
+  const handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
+    setShowTooltip(false);
+    setNodePosition(null);
+  };
+
+  const tailLength = tailLengths[tooltipProps.size || 'm'];
+
+  const updateTooltipStyles = useCallback(() => {
+    if (nodePosition && tooltipRef.current) {
+      const rect = nodePosition.getBoundingClientRect();
+      const tooltipRect = tooltipRef.current.getBoundingClientRect();
+      const direction = document.documentElement.dir || 'ltr'; // Default to 'ltr' if not specified
+
+      let styles: React.CSSProperties = {};
+
+      // Calculate the tooltip position based on the direction and its tail length
+      switch (tooltipProps.direction) {
+        case 'onRight':
+          styles = {
+            top: `${
+              rect.top +
```

Comment:
This math looks repeated / complicated with `onLeft` (maybe other repetition too?), could we extract it?

Reply:
Lines 74-77 and 105-109 are exactly the same, yeah? I think just at least pulling out exactly copied math would be good, not sure if you need to refactor to fancily do all math with single helper or anything.